alns/stop/tests/test_no_improvement.py

```diff
@@ -66,28 +72,21 @@ def test_stop_after_max_iterations():
         assert_(stop(rnd, Zero(), Zero()))
 
 
-def test_counter_value_at_stop():
-    """
-    Test if the counter value is equal to the passed-in max iterations
-    at the first stopping occurence.
-    """
-    stop = NoImprovement(100)
-
-    while not stop(RandomState(), Zero(), Zero()):
-        pass
-
-    assert_equal(stop._counter, 100)
-
-
-def test_reset_counter():
+def test_n_max_iterations_with_single_improvement():
     """
-    Test if the counter is reset when a new best solution is encountered.
+    Test if setting max_iterations to N correctly stops with a sequence
+    of solutions, where the third solution is improving and the other solutions
+    are non-improving. The first N+2 iterations should not stop. Beyond that,
+    the criterion should stop.
     """
     stop = NoImprovement(100)
     rnd = RandomState()
 
-    for _ in range(5):
-        assert_(not stop(rnd, One(), One()))
+    for _ in range(2):
+        assert_(not stop(rnd, One(), Zero()))
 
-    assert_(not stop(rnd, Zero(), One()))
-    assert_equal(stop._counter, 0)
+    for _ in range(100):
+        assert_(not stop(rnd, Zero(), Zero()))
+
+    for _ in range(100):
+        assert_(stop(rnd, Zero(), Zero()))
```

Comment:
Comment: this is really minor, but since the function mentions `n` as an arbitrary parameter, maybe set this up using pytest's parametrize?

Reply:
And similarly for the other test function with `n` in its name (just above this one).